x-pack/filebeat/input/gcppubsub/input.go

```diff
@@ -209,30 +213,67 @@ func (in *pubsubInput) run() error {
 	ctx, cancel := context.WithCancel(in.workerCtx)
 	defer cancel()
 
-	client, err := in.newPubsubClient(ctx)
-	if err != nil {
-		return err
+	numGoRoutines := in.Subscription.NumGoroutines
+	var workerWg sync.WaitGroup
+
+	for ctx.Err() == nil {
+		workers, err := in.workerSem.AcquireContext(numGoRoutines, ctx)
```

Comment:
I'm not sure we should proceed the approach of adding multiple pub/sub readers (pending resolution to previous comments), but if we do, then:

I think this can be simplified to not use a semaphore. A plain for loop (e.g. `for i := 0; i < number of readers; i++ { go runSinglePubSubClient() }`) seems to achieve the same result given that if any one "worker" fails they all stop due to `cancel()`.